pkg/apis/batch/validation/validation.go

```diff
@@ -564,13 +568,16 @@ func validateConcurrencyPolicy(concurrencyPolicy *batch.ConcurrencyPolicy, fldPa
 	return allErrs
 }
 
-func validateScheduleFormat(schedule string, timeZone *string, fldPath *field.Path) field.ErrorList {
+func validateScheduleFormat(schedule, oldSchedule string, timeZone *string, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	if _, err := cron.ParseStandard(schedule); err != nil {
 		allErrs = append(allErrs, field.Invalid(fldPath, schedule, err.Error()))
 	}
+	if !strings.Contains(oldSchedule, "TZ") && strings.Contains(schedule, "TZ") {
+		allErrs = append(allErrs, field.Invalid(fldPath, schedule, "cannot use TZ or CRON_TZ in schedule, use timeZone instead"))
+	}
 	if strings.Contains(schedule, "TZ") && timeZone != nil {
-		allErrs = append(allErrs, field.Invalid(fldPath, schedule, "cannot use both timeZone field and TZ or CRON_TZ in schedule"))
+		allErrs = append(allErrs, field.Invalid(fldPath, schedule, "cannot use TZ or CRON_TZ in schedule, use timeZone only"))
```

Comment:
From the logic I see, this error will only appear when setting `timeZone` for an existing CronJob that already uses the legacy mechanism.
(if that's right, we could add a code comment to that effect).
```suggestion
		allErrs = append(allErrs, field.Invalid(fldPath, schedule, "cannot define legacy TZ or CRON_TZ when also specifying timeZone; use timeZone only"))
```
?

Reply:
The goal afaict is to get them off TZ/CRON_TZ entirely, so not even mentioning this is legacy, b/c it's not, it was never supported.